lib/common.js

```diff
@@ -262,10 +262,19 @@ function mapVm(vm, source, cb) {
         if (vm.tags['smartdc_role'] || vm.tags['manta_role']) {
             has_core_tag = true;
         }
+        /*
```

Comment:
Is there an easy way to identify whether the vm's owner is poseidon at this point? If so, we should just add the manta_role to the groups array here rather than in the endpoint handler code. That way, we only have to do it once per vm, rather than every time we hit the endpoint.

If there's no easy way to identify poseidon here, the existing structure is fine with me -- just let me know.

Reply:
I had been trying to find one, and as far as I can tell, no. The owner_uuid is present, but uuid of poseidon is different in each installation. The login comes from the request body (really, the certificate that's passed by the client). I was looking for how to get access to the account cache, but from this point in the code it doesn't look possible, unless I go somewhere else and pass the cache into it, which may require several steps of passing that around. I'm not sure that's worth it. The only other option that I could think of to get it at that point is to look it up in UFDS directly.

Instead, I decided to follow how  `has_core_tag` does it, where the value gets added indiscriminately, and the account name is checked later in metrics. (see https://github.com/joyent/triton-cmon/blob/5b0974cbb150575469303a5b37f0ec7c833b1c32/lib/endpoints/metrics.js#L129-L131). I figured if that's already how it's being done, it wouldn't be so terrible to follow suit.

If there is a way though, it would be nice to fix both of these.

Reply:
Ah, this all makes sense. I agree that it's probably not worth it to do all of that work right now -- if performance turns out to be a problem (which I don't think it will be), we can handle it later.